### PR TITLE
Enable debug mode on BUILD=debug

### PR DESCRIPTION
**Pull request by wilzbach**
I expected `make -f posix.mak BUILD=debug` to work out of the box and generate a debug build of the compiler (with debug info!).

### CODE REVIEW

**Comment by CyberShadow** [comment]
I'm not sure these are equivalent. CC @yebblies

**Comment by wilzbach** [comment]
> I'm not sure these are equivalent. CC @yebblies

`ENABLE_DEBUG` adds the debug flags:

```makefile
 ifdef ENABLE_DEBUG
 CXXFLAGS += -g -g3 -DDEBUG=1 -DUNITTEST		 CXXFLAGS += -g -g3 -DDEBUG=1 -DUNITTEST
 DFLAGS += -g -debug -unittest		 DFLAGS += -g -debug -unittest
 endif		 endif
```

whereas `BUILD=debug` just changes the folder (from what I can tell).

**Comment by Burgos** [comment]
Yes, I was surprised by this behavior the other day as well.

File: src/posix.mak [hunk 1]
```diff
@@ -27,6 +27,7 @@ ifneq ($(BUILD),release)
     ifneq ($(BUILD),debug)
         $(error Unrecognized BUILD=$(BUILD), must be 'debug' or 'release')
     endif
+	ENABLE_DEBUG := 1
```
Comment (Burgos) [inline src/posix.mak]: tabs vs. spaces as in the rest of the file.

**Comment by WalterBright** [comment]
Please add documentation about what this does to the makefile, not just here.

File: src/posix.mak (hunk 1 above)
Comment (wilzbach) [inline src/posix.mak]: There's no consistent style:

image

image

image


...

**Comment by wilzbach** [comment]
> Please add documentation about what this does to the makefile, not just here.

This Makefile was more or less undocumented, so I started an attempt to document it.

**Comment by wilzbach** [comment]
As no one objected for the last two months, can we move forward with this trivial improvement?

**Comment by JinShil** [comment]
I ran into this confusion just the other day, so this PR is most welcome.  LGTM.

File: src/posix.mak [hunk 2]
```diff
@@ -1,3 +1,31 @@
+################################################################################
+# Important variables:
+# --------------------
+#
```
Comment (WalterBright) [inline src/posix.mak]: The formatting is off on these comments - perhaps you're using tabs?

File: src/posix.mak [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+# HOST_CSS:				Host C++ compiler to use (g++,clang++)
+# HOST_DMD: 			Host D compiler to use for bootstrapping
+# AUTO_BOOTSTRAP:		Enable auto-boostrapping by download a stable DMD binary
```
Comment (WalterBright) [inline src/posix.mak]: s/download/downloading/

File: src/posix.mak [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+# INSTALL_DIR:			Installation folder to use
+#
+################################################################################
+# Build modes:
+# ------------
+# BUILD: release (default) | debug (enabled a build with debug symbols)
```
Comment (WalterBright) [inline src/posix.mak]: Probably ought to say that this is intended to be set on the command line

**Pull request merged**